scripts/release.ts

```diff
@@ -0,0 +1,265 @@
+        .demandOption("pr");
+    },
+    handler: publish,
+  }).argv;
+
+function init(args) {
+  preflight();
+
+  const diff = diffJson();
+
+  // Start a release branch
+  // Convention borrowed from https://github.com/w3c/webref/blob/60ebf71b9d555c523975cfefb08f5420d12b7293/tools/prepare-release.js#L164-L165
+  const releaseBranch = `release-${new Date()
+    .toISOString()
+    .replace(/[\-T:\.Z]/g, "")}`;
+  logger.info(`Starting release branch ${releaseBranch}`);
+  const branchCmd = `git branch ${releaseBranch} HEAD`;
+  run(branchCmd);
+
+  // Check out release branch
+  logger.info(`Checking out release branch ${releaseBranch}`);
+  const checkoutCmd = `git checkout ${releaseBranch}`;
+  logger.debug(checkoutCmd);
+  execSync(checkoutCmd);
+
+  // Bump version (no tag)
+  const bumpCmd = `npm version --no-git-tag-version ${args.semverlevel}`;
+  logger.info("Bumping version number");
+  logger.debug(bumpCmd);
+  execSync(bumpCmd, { cwd: packages["web-features"], stdio: "inherit" });
+  const { version } = JSON.parse(
+    readFileSync(join(packages["web-features"], "package.json"), {
+      encoding: "utf-8",
+    })
+  );
+
+  // Commit
+  logger.info("Committing version bump");
+  const commitMessage = `Increment ${args.semverlevel} version to v${version}`;
+  const commitCmd = `git commit --all --message="${commitMessage}"`;
+  run(commitCmd);
+
+  // Push release branch
+  logger.info("Pushing release branch");
+  const pushCmd = `git push --set-upstream origin ${releaseBranch}`;
+  run(pushCmd);
+
+  // Create PR
+  logger.info(`Creating PR for ${version}`);
+  const title = `📦 Release web-features@${version}`;
+  const reviewer = "ddbeck";
+  const bodyFile = fileURLToPath(
+    new URL("release-pull-description.md", import.meta.url)
+  );
+
+  const temporaryDir = mkdtempSync(join(tmpdir(), "pull-request-"));
+  const temporaryBodyFile = join(temporaryDir, "body.md");
+  copyFileSync(bodyFile, temporaryBodyFile);
+  appendFileSync(temporaryBodyFile, ["```diff", diff, "```"].join("\n"));
+
+  const pullRequestCmd = [
+    "gh pr create",
+    `--title="${title}"`,
+    `--reviewer="${reviewer}"`,
+    `--body-file=${temporaryBodyFile}`,
+    `--base="main"`,
+    `--head="${releaseBranch}"`,
+    // `--repo="ddbeck/feature-set"`, // Uncomment this line to point this at a fork for testing
+  ].join(" ");
+  run(pullRequestCmd);
+
+  rmSync(temporaryDir, { recursive: true });
+}
+
+function update(args) {
+  // TODO: Rebase
+  // TODO: Run `npm run build`
+  // TODO: Commit results of `npm run build`
+  // TODO: Update PR diff
+  throw Error("Not implemented");
+}
+
+function publish(args) {
+  // TODO: Run `npm publish …`
+  throw Error("Not implemented");
+}
+
+function run(cmd: string) {
+  execSync(cmd, { stdio: "inherit" });
+}
+
+function diffJson(): string {
+  const temporaryDir = mkdtempSync(join(tmpdir(), "web-features-"));
+
+  execSync("npm install web-features", {
+    cwd: temporaryDir,
+    encoding: "utf-8",
```

Comment:
Is `encoding` needed in other calls to `execSync` as well?

Reply:
Really, it's superfluous here. Without a `stdio` option (or capturing the result of `execSync` and doing something with it), the output of `npm install web-features` disappears. I ought to either drop the encoding or let it rip with `stdio: "inherit"`. I've decided to go with the latter, at least for now.

As it stands, this whole thing is mostly a fancy shell script rather than a quiet, well-behaved standalone utility. If we find this thing performs reliably, we can suppress the output from the commands it runs, or make it toggle-able with a flag.